app/src/main/AndroidManifest.xml

```diff
@@ -44,8 +44,9 @@
         </receiver>
 
         <service
-                android:name=".player.MainPlayer"
-                android:exported="false">
+            android:name=".player.MainPlayer"
+            android:exported="false"
+            android:foregroundServiceType="mediaPlayback">
```

Comment:
The documentation doesn't say what effect this has. Are you just future-proofing the code?

Reply:
<sup>Obviously yes, as I want to make 0.20.0 as unstable as I possibly can without anyone knowing... just joking ;-)</sup>
This line is needed due to [this line](https://github.com/TeamNewPipe/NewPipe/pull/3178/files#diff-bd75c3fe4ebfdaa89acd9b52800338bbR165) in `NotificationUtil`: basically, on newer android versions, when starting a foreground service you can tell the system what its purpose is, so I did just that (also, on newer android versions, starting a foreground service without providing the type is deprecated). But in order to have a service of a specific type, it has to be declared as such in the manifest.

Reply:
> you can tell the system what its purpose is

Does this have any effect yet, though? Or is this something Android will use in the future so they simply introduced it now?

Reply:
For other typed of foreground services it has effects (e.g. storage can't be accessed in a foreground service without a storage service type, and other permission related things). For media playback I don't think it has, but it is still a flag that can be used by the system to optimize resource allocation, I think.